Lager 2.0 support

redgoattea opened the pull request:
Update the backend to be compatible with the Lager 2.0 Backend API changes.
- uses new lager formatter to allow customization of the log message
- compatible with Syslog style log level comparison flags
- implements lager config_to_id callback to enable create gen_event handler id based on eventlog source

redgoattea committed:
Support lager 2.0 message format

The Location string used in Lager < 2.0 was either
  io_lib:format("~p ", [Pid])
-or-
  io_lib:format("~p@~p:~p:~p ", [Pid, Module, Function, Line])

The default format creates similar log entries while not needing
to do a string flatten to add the log message.

redgoattea committed:
Allow for custom formats via configuration

when using lager 2.0 or newer.

redgoattea committed:
Support optional lager config_to_id callback

redgoattea committed:
Support syslog style log level comparison flags

redgoattea committed:
bump version to 1.0.0

redgoattea committed:
corrected record error in code_change

benjaminplee commented on src/lager_nt_eventlog_backend.erl:
```diff
@@ -4,19 +4,30 @@
-            {ok, #state{level=lager_util:level_to_num(Level),
-                        handle = Handle}};
+    init([Source, Level, {lager_default_formatter, ?DEFAULT_FORMAT}]);
+init([Source, Level, {Formatter, FormatterConfig}]) ->
```
I would note this in the README; both the default format as well as the option to include a custom one.

benjaminplee commented on src/lager_nt_eventlog_backend.erl:
```diff
@@ -46,5 +70,36 @@ terminate(_Reason, #state{handle = Handle}) ->
+            Mask
+    end.
+
+parse_level(Level, 2) ->
```
This is ugly (the 2 and 1)

benjaminplee commented on src/lager_nt_eventlog_backend.erl:
```diff
@@ -46,5 +70,36 @@ terminate(_Reason, #state{handle = Handle}) ->
 
 %% @private
+code_change("0.9.0", {state, Handle, Level}, _Extra) ->
+    {ok, #state{id = ?MODULE, handle = Handle, level = Level, format_config = lager_default_formatter, format_config = ?DEFAULT_FORMAT }};
```
Should the old Level be parsed?

redgoattea committed:
make parse function names more explicit

rather than using the api version like a magic number who's meaning gets lost.

redgoattea committed:
add custom formatting section to readme

redgoattea committed:
reword custom formatting section

to give better specifics on the requirements for the format/2 callback
and make it more explicit that the defaults only apply if the entire formatter tuple is missing

benjaminplee commented:
Looks good
